src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+    /// let mut v = Vob::new();
+    /// v.push(false);
+    /// v.push(true);
+    /// let mut iterator = v.iter_unset_bits();
+    /// assert_eq!(iterator.next(), Some(0));
+    /// assert_eq!(iterator.next(), None);
+    /// ```
+    pub fn iter_unset_bits(&self) -> IterUnsetBits<T> {
+        IterUnsetBits {
+            vob: self,
+            range: 0..self.len
+        }
+    }
+
+    /// Return an iterator over the underlying storage blocks. The last block is guaranteed to have
+    /// "unused" bits (i.e. those past `self.len()`) set to 0.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let v1 = Vob::from_elem(10, true);
+    /// assert_eq!(v1.iter_storage().next(), Some((1 << 10) - 1));
+    /// let v2 = Vob::from_elem(129, true);
+    /// assert_eq!(v2.iter_storage().last(), Some(1));
+    /// ```
+    pub fn iter_storage(&self) -> StorageIter<T> {
+        StorageIter{iter: self.vec.iter()}
+    }
+
+    /// Resizes the Vob in-place so that `len` is equal to `new_len`.
+    ///
+    /// If `new_len` is greater than `len`, the Vob is extended by the difference, with each
+    /// additional slot filled with `value`. If `new_len` is less than `len`, the vob is simply
+    /// truncated.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v = Vob::new();
+    /// v.push(false);
+    /// v.resize(129, true);
+    /// assert_eq!(v.len(), 129);
+    /// assert_eq!(v.get(0), Some(false));
+    /// assert_eq!(v.get(128), Some(true));
+    /// ```
+    pub fn resize(&mut self, new_len: usize, value: bool) {
+        if new_len <= self.len {
+            self.truncate(new_len);
+            return
+        }
+        if value && self.len > 0 {
+            // If we're resizing with trues, we need to extend the last block with true bits. We
+            // can rely on mask_last_block to trim any unwanted bits we add in this process.
+            let b = block_offset::<T>(self.len);
+            let v = self.vec[b];
+            self.vec[b] = v | (T::max_value() << (self.len % bits_per_block::<T>()));
+        }
+        let nb = blocks_required::<T>(new_len);
+        self.vec.resize(nb, if value { T::max_value() } else { T::zero() });
+        self.len = new_len;
+        self.mask_last_block();
+    }
+
+    /// Appends all elements in a slice to the Vob.
+    ///
+    /// Iterates over the slice `other` and appends elements in order.
+    ///
+    /// Note that this function is same as extend except that it is specialized to work with slices
+    /// instead. If and when Rust gets specialization this function will likely be deprecated (but
+    /// still available).
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true];
+    ///     v.extend_from_slice(&vec![false, true]);
+    ///     assert_eq!(v, vob![true, false, true]);
+    /// }
+    pub fn extend_from_slice(&mut self, other: &[bool]) {
```

Comment:
Could also be optimised, although a little trickier.

Easy case, the length of `self` is right up to capacity. Then it's just a matter of using the underlying `vec`s extend method.

[not for this PR]

Reply:
Agreed.